src/ShareReceived.php

```diff
@@ -29,14 +30,12 @@ public function __construct(
     /**
      *
      * @return string
-     * @throws InvalidResponseException
+     * @throws TooEarlyException
      */
     public function getId(): string
     {
-        return empty($this->shareReceived->getId())
-            ? throw new InvalidResponseException(
-                "Invalid Id '" . print_r($this->shareReceived->getId(), true) . "'"
-            )
+        return ($this->shareReceived->getId() === null)
+            ? throw new TooEarlyException()
```

Comment:
Is that the correct exception? It might be that we are checking too-early and an auto-accept did not happen yet, or there is no auto-accept and the user has to manually accept the share AKA "set it to sync" in the webUI